connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractHerder.java

```diff
@@ -184,7 +184,31 @@ public void resumeConnector(String connector) {
         configBackingStore.putTargetState(connector, TargetState.STARTED);
     }
 
-    @Override
+@Override
+public Map<String, String> maskCredentials(String connName, Map<String, String> config) {
+    Connector connector = getConnector(config.get(ConnectorConfig.CONNECTOR_CLASS_CONFIG));
+    if (connector == null) {
+        throw new NotFoundException("No status found for connector " + connName);
```

Comment:
This exception message doesn't seem to match the method. Perhaps the following?

    throw new NotFoundException("Unable to find connector " + connName)